src/app/user/[id]/setting/edit/UserEdit.controller.tsx

```diff
@@ -1,37 +1,82 @@
 "use client";
 
-import { useForm } from "react-hook-form";
+import { FieldErrors, useForm } from "react-hook-form";
 import UserEditView from "./UserEdit.view";
+import { TopNavigator } from "@/components/navigators/TopNavigator";
+import { GoBackButton } from "@/components/navigators/TopNavigator/components";
+import { UserEditData } from "./UserEdit.types";
+import useUserEditModel from "./UserEdit.model";
+import { useEffect } from "react";
 
 interface UserEditControllerProps {
-  userId: string;
+  userData: UserEditData;
 }
 
-interface UserData {
-  nickname: string;
-  sex: "male" | "female";
-  age: number;
-  height: number;
-  weight: number;
-}
+// userId 는 추후 유저의 데이터를 불러올때 사용
+const UserEditController = ({ userData }: UserEditControllerProps) => {
+  const { isCheckedNickname, setIsCheckedNickname } = useUserEditModel();
+  const { register, handleSubmit, watch, formState, getValues } = useForm<UserEditData>({
+    defaultValues: userData,
+  });
+  const { errors } = formState;
 
-const FORM_DEFAULT_VALUE: UserData = {
-  nickname: "",
-  sex: "male",
-  age: 12,
-  height: 182,
-  weight: 150,
-};
+  const newNickname = watch("nickname");
+  const selectedSex = watch("sex");
 
-const UserEditController = ({ userId }: UserEditControllerProps) => {
-  const {} = useForm<UserData>({
-    mode: "onChange",
-    defaultValues: FORM_DEFAULT_VALUE,
-  });
+  useEffect(() => {
+    if (userData.nickname === newNickname) {
+      setIsCheckedNickname(true);
+      return;
+    }
+
+    if (isCheckedNickname) {
+      setIsCheckedNickname(false);
+    }
+  }, [newNickname]);
```

Comment:
[ P4 ]
useEffect 보다는 useMemo 를 사용해서 checkedNickname 을 관리하면 어떨까 싶네요.

Reply:
해당 부분에서도 값을 변동시켜줘야하고 여러 부분에서 관리가 이루어지다보니 적용이 어렵네요 .. ㅠ

```suggestion
  const handleCheckSameNickName = () => {
    const newNickname = getValues("nickname");

    // TODO - API 연결로 인해 중복 닉네임 체크후 없는 닉네임이라면 통과처리

    // if(){
    //   console.log(newNickname);
    // }

    setIsCheckedNickname(true);
  };
```